controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -257,10 +257,17 @@ func (r *DataScienceClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 
 	// finalize reconciliation
+	operatorReleaseVersion, err := cluster.SetRelease(r.Client)
+	if err != nil {
+		r.Log.Error(err, "failed to get operator release version")
+		return ctrl.Result{}, err
+	}
 	instance, err = status.UpdateWithRetry(ctx, r.Client, instance, func(saved *dsc.DataScienceCluster) {
 		status.SetCompleteCondition(&saved.Status.Conditions, status.ReconcileCompleted, "DataScienceCluster resource reconciled successfully")
 		saved.Status.Phase = status.PhaseReady
+		saved.Status.Release = *operatorReleaseVersion
```

Comment:
do we need this Release to be set on both DSCI and DSC?

Reply:
I think that it is good to have it on both, considering that these resources may or may not exist, independently. It also has the potential to help resolve conflicts in the future when custom upgrade logic might be necessary.

Reply:
Yes I added them mostly so that we can add upgrade logic to follow-up after this change